emukungu/IOT
# feat(optimization): cache data from ThingsSpeak API

#### What does this PR do?
- Optimize the process of retrieving data from the ThingsSpeak API.

#### Description of Task to be completed?
- Consume the ThingsSpeak API to retrieve weather data
- Use` redis` to cache data locally for a period of `2minutes`, then the next request will reload the page with the updated data from the api.

#### How should this be manually tested?
1. Check out to the branch `ft-cache-data`
2. Create a virtual environment for the project, run `virtual env`
3. Install dependencies using `pip install -r requirements.txt`
4. On terminal, run command `python manage.py runserver`

#### Any background context you want to provide?
- N/A

#### What are the relevant pivotal tracker stories?
N/A

#### Postman Documentation
- N/A

#### Screenshots (If applicable)
<img width="1440" alt="Screen Shot 2019-08-08 at 8 24 02 PM" src="https://user-images.githubusercontent.com/9901086/62724082-d3fd0000-ba1a-11e9-86e7-e68ba8cd95b3.png">
